run-servers.py

```diff
@@ -56,16 +56,19 @@ def run(i):
     lock.release()
     # join if not the first node
     if len(procs) != 0: 
-        cmd += ["-joinAddr", f"localhost:{LEADER_PORT + procs[0][0]}"]
```

Comment:
tbh this doesn't ensure that procs[0][0] is a leader as it is randomized. we dont really have a good way to "know" who is the leader as a client. 
alternative is to maybe scan the recent logs of each server but thats too much for now lol

Reply:
not sure if the join req needs to be to the leader, but didn't test